python_modules/dagster/dagster/_core/instance/__init__.py

```diff
@@ -1900,6 +1905,108 @@ def get_event_records(
         """
         return self._event_storage.get_event_records(event_records_filter, limit, ascending)
 
+    @public
+    @traced
+    def get_materialization_records(
+        self,
+        records_filter: Optional[Union[AssetKey, "AssetRecordsFilter"]] = None,
+        limit: int = 10000,
+        cursor: Optional[str] = None,
+        ascending: bool = False,
```

Comment:
are we sure we want to inherit our always ordered you just pick up or down set-up? I feel like we have seen queries that we wanted to un-order but this abstraction setup makes untenable

Reply:
I'm not sure how we'd apply the cursor filter without it...

At the very least, we'd need to encode it into the cursor string.

Reply:
ok ya thats right 

this upper layer makes sense to be ordered, i think our problems with ordering-too-much come from the lower abstractions in query building